### PR TITLE
#7427: Restore some sidePanel messaging usage; move code to clarify usage

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Fixes https://github.com/pixiebrix/pixiebrix-extension/issues/7427
- Follows #7354

This PR moves a lot of code around to clarify its usage (e.g. _to_ `/contentScript/*` if it's contentScript-only). I'll highlight the actual code changes in the comments.

## Checklist

- [ ] Add tests
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer: @fungairino

### CODE REVIEW

File: src/contentScript/sidebarController.tsx [hunk 1]
```diff
@@ -37,22 +37,59 @@ import type {
...
+async function isSidePanelOpenMv3(): Promise<boolean> {
+  if (isSidePanelOpenSync() === false) {
+    return false;
+  }
+
+  try {
+    await messenger(
+      "SIDEBAR_PING",
+      { retry: false },
+      await getSidebarTargetForCurrentTab(),
```
Comment (fregante) [inline src/contentScript/sidebarController.tsx:86]: Change:

- `isSidePanelOpen` was split into two version:
  - this one contentScript-only, under `/contentScript/*`
  - the other one from any other context, in `/utils/sidePanelUtils.ts`

File: src/utils/sidePanelUtils.ts [hunk 2]
```diff
@@ -15,68 +15,38 @@
...
-  return {
-    page: getSidebarPath(frame.tabId),
-  };
-}
-
-export async function isSidePanelOpen(): Promise<boolean> {
-  expectContext(
+  forbidContext(
     "contentScript",
-    "isSidePanelOpen only works from the same content script for now",
```
Comment (fregante) [inline src/utils/sidePanelUtils.ts:56]: This contentScript-only API was moved under `/contentScript/`

The any-context version is below

File: src/utils/sidePanelUtils.ts [hunk 3]
```diff
@@ -96,7 +64,7 @@ export async function _openSidePanel(tabId: number): Promise<void> {
     // The following code silences that error iff the side panel is already open.
     if (
       getErrorMessage(error).includes("user gesture") &&
-      (await isSidePanelOpen())
+      (await isSidePanelOpen(tabId))
```
Comment (fregante) [inline src/utils/sidePanelUtils.ts:67]: This was the confusing/wrong usage. Splitting and moving the functions to context-specific folders fixes the issue in title.

File: src/tsconfig.strictNullChecks.json [hunk 4]
```diff
@@ -480,6 +480,7 @@
     "./types/stringTypes.ts",
     "./types/swagger.ts",
     "./urlConstants.ts",
+    "./utils/sidePanelUtils.ts",
```
Comment (grahamlangford) [inline src/tsconfig.strictNullChecks.json:483]: 💜

**Review by BLoe — APPROVED** [review]

**Pull request merged**